pkg/api/store/auth/store.go

```diff
@@ -45,15 +54,50 @@ func (s *Store) Update(apiContext *types.APIContext, schema *types.Schema, data
 	}
 
 	kind := convert.ToString(authType)
-	fields := TypeToFields[kind]
-	for _, field := range fields {
-		if val, ok := data[field]; ok {
-			val := convert.ToString(val)
-			if err := common.CreateOrUpdateSecrets(s.Secrets, val, strings.ToLower(field), strings.ToLower(kind)); err != nil {
-				return nil, fmt.Errorf("error creating secret for %s:%s", kind, field)
+	fields, ok := TypeToFields[kind]
+	subFields, subOk := SubTypeToFields[kind]
+	if !ok && !subOk {
+		return s.Store.Update(apiContext, schema, data, id)
+	}
+
+	var err error
+	if ok {
```

Comment:
nit: we don't need this check and the check for `if subOk`

Reply:
its an or situation tho, a provider could have secrets in the fields or subfields and since i had both in case the authprovider has no secrets i wrapped both or sections to be safe